app/components-react/widgets/ChatBox.tsx

```diff
@@ -111,15 +111,44 @@ export class ChatBoxModule extends WidgetModule<IChatBoxState> {
       },
       muted_chatters: metadata.textarea({ label: $t('Muted Chatters') }),
     };
+
+    if (this.config.useNewWidgetAPI) {
+      return {
+        ...result,
+        alert_enabled: metadata.switch({
+          label: $t('Chat Notifications'),
+          tooltip: $t('Trigger a sound to notify you when there is new chat activity'),
+        }),
+      };
+    }
+
+    return result;
+  }
+
+  // TODO: move these to widget module in a more generic way
+  protected patchAfterFetch(resp: any) {
+    if (this.config.useNewWidgetAPI) {
+      const data = {
+        settings: resp.data.settings.global,
+      };
+
+      return data;
+    }
+
+    return resp;
   }
 
   // The server sends and recieves these duration fields at different precision
   protected patchBeforeSend(settings: IChatBoxState['data']['settings']) {
-    return {
+    const obj = {
       ...settings,
       message_hide_delay: Math.floor(settings.message_hide_delay / 1000),
       message_show_delay: Math.floor(settings.message_show_delay / 1000),
     };
+
+    if (this.config.useNewWidgetAPI) {
+      return { global: obj };
```

Comment:
do we no longer need to do math on the message delays in the new api?

Reply:
We're passing the same object, in which we already did the conversion. As far as I know that is still required.

Reply:
good point, but it does look like we're not returning the obj in case the flag is false

Reply:
Fixed in #5197, might be dead code now.